src/views/LogView.ts

```diff
@@ -0,0 +1,214 @@
+   */
+  private async resolveLazyLogs(): Promise<void> {
+    // Reads all stored logs and checks the action that will be taken
+    this._lazyLogs.forEach(async (log) => {
+      switch (log.command) {
+        case Commands.createLog:
+          // Creates the log
+          await this.createLog(log.label, log.message!, log.description);
+          break;
+        case Commands.disposeTab:
+          // Disposes the tab
+          await this.disposeTab(log.label, log.description);
+          break;
+      }
+    });
+
+    // Resets the array
+    this._lazyLogs = [];
+  }
+
+  /**
+   * This function is responsible for checking the type of tool that is
+   * emitting the log: truffle, ganache or dashboard
+   *
+   * @param label - Enum with the representation of the log type
+   */
+  private async getTool(label: OutputLabel): Promise<string> {
+    let tool: string;
+
+    switch (label) {
+      case OutputLabel.ganacheCommands:
+        // Ganache
+        tool = Constants.panels.log.tool.ganache;
+        break;
+      case OutputLabel.dashboardCommands:
+        // Dashboard
+        tool = Constants.panels.log.tool.dashboard;
+        break;
+      // Truffle
+      default:
+        tool = Constants.panels.log.tool.truffle;
+        break;
+    }
+
+    return tool;
+  }
+
+  /**
+   * This function is responsible for retrieving the html from within the file
+   * and returning it to the web provider so that it can be printed.
+   */
+  private async getHtmlForWebview(): Promise<string> {
+    // Retrieves the HTML from the file
+    const file = await fs.readFile(
+      this._context.asAbsolutePath(path.join('resources', 'logPanel', 'index.html')),
+      'utf8'
+    );
+
+    // Creates a virtual path so that assets from within the HTML can be loaded and return the HTML
+    const virtualPath = this._view?.webview.asWebviewUri(this._extensionUri).toString();
+    return file.replace(/{{root}}/g, virtualPath!);
+  }
+
+  /**
+   * This event is responsible for checking the visibility of the log panel
+   */
+  private async onViewVisibilityChanged() {
+    // Check if the view is visible
+    if (this._view?.visible) {
+      // Sends the message to the client so that the log history can be retrieved and printed
+      this._view?.webview.postMessage({command: 'get.history'});
+      // After retrieving the history, resolve new logs
+      await this.resolveLazyLogs();
+    }
+  }
+
+  /**
+   * This event is responsible for disposing the log panel
+   */
+  private async onViewDisposed() {
+    this._view = undefined;
+  }
+}
+
+/**
+ * This function is responsible for registering the log view
+ *
+ * @param context - An extension context is a collection of utilities private to an extension
+ */
+export function registerLogView(context: vscode.ExtensionContext): vscode.Disposable {
+  const logView = new LogView(context);
+
+  // Registers the command responsible for creating the logs
+  vscode.commands.registerCommand(
```

Comment:
Not sure if/when this can be an issue, but the `Disposable` returned by `registerCommand` is not included in `context.subscriptions`. See for example `extension.ts::.activate`